qcodes/instrument_drivers/signal_hound/USB_SA124B.py

```diff
@@ -1,166 +1,327 @@
+
+    def _update_trace(self) -> None:
+        """
+        Private method to sync changes of the
+        frequency axis to the setpoints of the
+        trace parameter.
+        """
+        sweep_info = self.QuerySweep()
+        self.npts._save_val(sweep_info[0])
+        self.trace.set_sweep(*sweep_info)
+
+    def _sync_parameters(self) -> None:
+        """
+        Sync parameters consists of five parts
+            1. Center span configuration (freqs and span)
+            2. Acquisition configuration
+                lin-scale/log-scale
+                avg/max power
+            3. Configuring the external 10MHz reference
+            4. Configuration of the mode that is being used
+            5. Configuration of the tracking generator (not implemented)
+                used in VNA mode
+
+        Sync parameters sets the configuration of the instrument using the
+        parameters specified in the Qcodes instrument.
+        """
+
+        # 1. CenterSpan Configuration
+        center = ct.c_double(self.frequency())
+        span = ct.c_double(self.span())
+        log.info('Setting device CenterSpan configuration.')
+
+        err = self.dll.saConfigCenterSpan(self.deviceHandle, center, span)
+        self.check_for_error(err)
+
+        # 2. Acquisition configuration
+        detectorVals = {
+            'min-max': ct.c_uint(self.hf.sa_MIN_MAX),
+            'average': ct.c_uint(self.hf.sa_AVERAGE)
+        }
+        scaleVals = {
+            'log-scale': ct.c_uint(self.hf.sa_LOG_SCALE),
+            'lin-scale': ct.c_uint(self.hf.sa_LIN_SCALE),
+            'log-full-scale': ct.c_uint(self.hf.sa_LOG_FULL_SCALE),
+            'lin-full-scale': ct.c_uint(self.hf.sa_LIN_FULL_SCALE)
+        }
+        detector = detectorVals[self.acquisition_mode()]
+        scale = scaleVals[self.scale()]
+
+        err = self.dll.saConfigAcquisition(self.deviceHandle, detector, scale)
+        self.check_for_error(err)
 
-    @classmethod
-    def default_server_name(cls, **kwargs):
-        return 'USB'
+        # 3. Reference Level configuration
+        log.info('Setting device reference level configuration.')
+        err = self.dll.saConfigLevel(
+            self.deviceHandle, ct.c_double(self.ref_lvl()))
+        self.check_for_error(err)
 
-    def openDevice(self):
+        # 4. External Reference configuration
+        if self.external_reference():
+            log.info('Setting reference frequency from external source.')
+            err = self.dll.saEnableExternalReference(self.deviceHandle)
+            self.check_for_error(err)
+
+        reject_var = ct.c_bool(self.reject_image())
+        log.info('Setting device Sweeping configuration.')
+        err = self.dll.saConfigSweepCoupling(
+            self.deviceHandle, ct.c_double(self.rbw()),
+            ct.c_double(self.vbw()), reject_var)
+        self.check_for_error(err)
+        self._parameters_synced = True
+
+    def configure(self) -> None:
```

Comment:
it is confusing to me that there are the following methods and it is not clear when should each of them be used and when more than one of them should be used: `configure`, `_prepare_measurement`, `_initialize`, `_update_trace`, `_sync_parameters` (well, the last two are clear). My suggestion is to remove `_prepare_measurement` method, and add docstring to `_initialize` - does it make sense?

Reply:
I think it makes sense to merge sync_parameters and _initialize and get rid of prepare_measurement which just calls those two.